mavros/include/mavros/setpoint_mixin.h

```diff
@@ -193,7 +200,46 @@ class TF2ListenerMixin {
 						tf_transform_cb(transform);
 					}
 					catch (tf2::LookupException &ex) {
-						ROS_ERROR_NAMED("tf2_buffer", "%s: %s", tf_thd_name.c_str(), ex.what());
+						ROS_ERROR_NAMED("tf2_buffer", "%s: %s", tf_thd_name_1.c_str(), ex.what());
+					}
+				}
+				rate.sleep();
+			}
+		});
+	}
+
+	/**
+	 * @brief start tf listener syncronized with mavros_msgs::Thrust publisher topic
+	 *
+	 * @param _thd_name  listener thread name
+	 * @param cbp        plugin callback function
+	 */
+	void tf2_sync_start(const char *_thd_name, void (D::*cbp)(const geometry_msgs::TransformStamped &, const mavros_msgs::Thrust::ConstPtr &))
```

Comment:
The only optimization I can think of is generalize the type of message we are syncing the tf listener with, instead of with the thrust only. That way we have a generalized way of syncing different types of topics with the transform.